src/engine/Bind.cpp

```diff
@@ -81,64 +81,95 @@ std::vector<QueryExecutionTree*> Bind::getChildren() {
+  }
+  auto localVocab = std::make_shared<LocalVocab>();
+  auto generator =
+      [](std::shared_ptr<LocalVocab> vocab, auto applyBind,
+         std::shared_ptr<const Result> result) -> cppcoro::generator<IdTable> {
+    for (IdTable& idTable : result->idTables()) {
+      co_yield applyBind(idTable, vocab.get());
+    }
+    std::array<const LocalVocab*, 2> vocabs{vocab.get(), &result->localVocab()};
+    *vocab = LocalVocab::merge(std::span{vocabs});
+  }(localVocab, std::move(applyBind), std::move(subRes));
+  return {std::move(generator), resultSortedOn(), std::move(localVocab)};
 }
 
 // _____________________________________________________________________________
 template <size_t IN_WIDTH, size_t OUT_WIDTH>
-void Bind::computeExpressionBind(
-    IdTable* outputIdTable, LocalVocab* outputLocalVocab,
-    const Result& inputResultTable,
+IdTable Bind::computeExpressionBind(
+    LocalVocab* outputLocalVocab,
+    ad_utility::SimilarTo<IdTable> auto&& inputIdTable,
+    const LocalVocab& inputLocalVocab,
     sparqlExpression::SparqlExpression* expression) const {
   sparqlExpression::EvaluationContext evaluationContext(
-      *getExecutionContext(), _subtree->getVariableColumns(),
-      inputResultTable.idTable(), getExecutionContext()->getAllocator(),
-      inputResultTable.localVocab(), cancellationHandle_, deadline_);
+      *getExecutionContext(), _subtree->getVariableColumns(), inputIdTable,
+      getExecutionContext()->getAllocator(), inputLocalVocab,
+      cancellationHandle_, deadline_);
 
   sparqlExpression::ExpressionResult expressionResult =
       expression->evaluate(&evaluationContext);
-
-  const auto input = inputResultTable.idTable().asStaticView<IN_WIDTH>();
-  auto output = std::move(*outputIdTable).toStatic<OUT_WIDTH>();
-
-  // first initialize the first columns (they remain identical)
-  const auto inSize = input.size();
-  output.reserve(inSize);
-  const auto inCols = input.numColumns();
-  // copy the input to the first numColumns;
-  for (size_t i = 0; i < inSize; ++i) {
-    output.emplace_back();
-    for (size_t j = 0; j < inCols; ++j) {
-      output(i, j) = input(i, j);
+  size_t inSize = inputIdTable.size();
+  size_t inCols = inputIdTable.numColumns();
+
+  auto output = [this, &inputIdTable, inSize, inCols]() {
+    if constexpr (std::is_const_v<
+                      std::remove_reference_t<decltype(inputIdTable)>>) {
+      const auto input = inputIdTable.template asStaticView<IN_WIDTH>();
+      auto output =
+          IdTable{getResultWidth(), getExecutionContext()->getAllocator()}
+              .template toStatic<OUT_WIDTH>();
+
+      // first initialize the first columns (they remain identical)
+      output.reserve(inSize);
+      // copy the input to the first numColumns;
+      for (size_t i = 0; i < inSize; ++i) {
+        output.emplace_back();
+        for (size_t j = 0; j < inCols; ++j) {
+          output(i, j) = input(i, j);
+        }
+        checkCancellation();
+      }
```

Comment:
This is basically
`auto output = input.clone(); output.addEmptyColumn(); return output`. You could use the opportunity to implement
the cloning without an optional cancellation handle etc.

Reply:
In particular, the clone is (and here should be) columnwise, so potentially much faster.